torch/csrc/utils/tensor_apply.cpp

```diff
@@ -83,11 +83,11 @@ Tensor & map2_(Tensor & self, const Tensor & x_, const Tensor & y_, PyObject* fn
   }
   if (x_.type() != self.type()) {
     throw TypeError("map2_: expected %s for argument 'x' (got %s)",
-        self.type().toString(), x_.type().toString());
+        self.type().toString().c_str(), x_.type().toString().c_str());
```

Comment:
you're a lucky man, temporary lifetime extends for the entirety of the expression lol